wpilibc/src/main/native/include/frc/ResistanceCalculator.h

```diff
@@ -0,0 +1,110 @@
+ * finding the resistance of a channel, use the calculate method with the
+ * battery voltage minus the voltage at the motor controller or whatever is
+ * plugged in to the PDP at that channel.</p>
+ */
+class ResistanceCalculator {
+ public:
+  static constexpr int kDefaultBufferSize = 250;
+  static constexpr double kDefaultRSquaredThreshold = 0.75;
+
+  /**
+   * Create a ResistanceCalculator to find the resistance of a channel using a
+   * running linear regression over a window. Must be updated with current and
+   * voltage periodically using the <code>Calculate</code> method.
+   *
+   * @param bufferSize The maximum number of points to take the linear
+   * regression over.
+   * @param krSquaredThreshold The minimum R² value (0 to 1) considered
+   * significant enough to return the regression slope instead of NaN. A lower
+   * threshold allows resistance to be returned even with noisier data.
+   */
+  ResistanceCalculator(int bufferSize, double krSquaredThreshold);
+
+  /**
+   * Create a ResistanceCalculator to find the resistance of a channel using a
+   * running linear regression over a window. Must be updated with current and
+   * voltage periodically using the <code>Calculate</code> method. <p>Uses a
+   * buffer size of 250 and an R² threshold of 0.5.</p>
+   */
+  ResistanceCalculator();
+
+  ~ResistanceCalculator() = default;
+  ResistanceCalculator(ResistanceCalculator&&) = default;
+  ResistanceCalculator& operator=(ResistanceCalculator&&) = default;
+
+  /**
+   * Update the buffers with new (current, voltage) points, and remove old
+   * points if necessary.
+   *
+   * @param current The current current
+   * @param voltage The current voltage
+   * @return The current resistance in ohms
+   */
+  units::ohm_t Calculate(units::ampere_t current, units::volt_t voltage);
+
+ private:
+  /**
+   * Buffers holding the current values that will eventually need to be
+   * subtracted from the sum when they leave the window.
+   */
+  wpi::circular_buffer<units::ampere_t> m_currentBuffer;
+  /**
+   * Buffer holding the voltage values that will eventually need to be
+   * subtracted from the sum when they leave the window.
+   */
+  wpi::circular_buffer<units::volt_t> m_voltageBuffer;
+  /**
+   * The maximum number of points to take the linear regression over.
+   */
+  int m_bufferSize;
+  /**
+   * The minimum R² value considered significant enough to return the
+   * regression slope instead of NaN.
+   */
+  double m_rSquaredThreshold;
+  /**
+   * Running sum of the past currents.
+   */
+  units::ampere_t m_currentSum;
+  /**
+   * Running sum of the past voltages.
+   */
+  units::volt_t m_voltageSum;
+  /**
+   * Running sum of the squares of the past currents.
+   */
+  units::unit_t<units::squared<units::current::amperes>> m_currentSquaredSum;
+  /**
+   * Running sum of the squares of the past voltages.
+   */
+  units::unit_t<units::squared<units::voltage::volts>> m_voltageSquaredSum;
```

Comment:
```suggestion
  units::unit_t<units::squared<units::volts>> m_voltageSquaredSum;
```
Consider `using volts_squared_t = ...;` typedef.